maestro-ios-driver/src/main/kotlin/xcuitest/XCTestDriverClient.kt

```diff
@@ -152,8 +152,9 @@ class XCTestDriverClient(
 
     fun inputText(
         text: String,
+        appId: String?,
```

Comment:
Let's not put a separate api call here for running app id. We have RunningApp struct which can help reducing these HTTP calls to 1.

Reply:
the thing is we also need to call getInstalledApps on simctl before doing this call, so it'd not be possible to do that completely on xctest driver side.

we could pass appIds list to the request, but that'd also look ugly imo.

Reply:
What weirdness you are referring to? We did that in viewHierarchy maybe be we can address there as well

Reply:
ok, I see. I don't see a big issue with extra network request here anyway since it is done on localhost and we can reuse this logic. 

but if we choose a pattern of resolving runningAppId on the xctest driver side I'm fine with doing this change.

I'd also deprecate runningAppId endpoint then

Reply:
>  I don't see a big issue with extra network request

Well, it's about okhttp. Doing too many calls does affect the connection pools and network latency. But since we haven't seen such reports I would say for me apart from this it also makes less complex. But open to discuss this and we can definitely deprecate runningAppId endpoint.


@berikv would appreciate your opinion about this as well.

Reply:
>  it is done on localhost and we can reuse this logic.
That's not always true.
If it is easy to switch to a similar setup as /viewHierarchy, I suggest we implement that. If otoh it is much more difficult, then lets keep it as in the PR now.

Reply:
thanks for the arguments guys! I think it makes sense to have it on the xctest driver side now

Reply:
fixed ✅